src/FeatureExtractor/Mobilenet.js

```diff
@@ -115,38 +118,33 @@ class Mobilenet {
       }
     }
 
-    if (this.modelLoaded) {
-      tf.tidy(() => {
-        const processedImg = imgToTensor(imgToAdd);
-        const prediction = this.mobilenetFeatures.predict(processedImg);
-
-        let y;
-        if (this.usageType === 'classifier') {
-          y = tf.tidy(() => tf.oneHot(tf.tensor1d([label], 'int32'), this.numClasses));
-        } else if (this.usageType === 'regressor') {
-          y = tf.tidy(() => tf.tensor2d([[label]]));
-        }
-
-        if (this.xs == null) {
-          this.xs = tf.keep(prediction);
-          this.ys = tf.keep(y);
-          this.hasAnyTrainedClass = true;
-        } else {
-          const oldX = this.xs;
-          this.xs = tf.keep(oldX.concat(prediction, 0));
-          const oldY = this.ys;
-          this.ys = tf.keep(oldY.concat(y, 0));
-          oldX.dispose();
-          oldY.dispose();
-          y.dispose();
-        }
-      });
-      if (callback) {
-        callback();
+    await this.ready;
+    tf.tidy(() => {
+      const processedImg = imgToTensor(imgToAdd);
+      const prediction = this.mobilenetFeatures.predict(processedImg);
+
+      let y;
+      if (this.usageType === 'classifier') {
+        y = tf.tidy(() => tf.oneHot(tf.tensor1d([label], 'int32'), this.numClasses));
+      } else if (this.usageType === 'regressor') {
+        y = tf.tidy(() => tf.tensor2d([[label]]));
```

Comment:
no need for tidy() here this actually won't do anything (there are no intermediate tensors)